samples/nrf9160/location/src/main.c

```diff
@@ -0,0 +1,290 @@
+				event_data->location.datetime.minute,
+				event_data->location.datetime.second,
+				event_data->location.datetime.ms);
+		}
+		printk("  Google maps URL: https://maps.google.com/?q=%.06f,%.06f\n\n",
+			event_data->location.latitude, event_data->location.longitude);
+		break;
+
+	case LOC_EVT_TIMEOUT:
+		printk("Getting location timed out\n\n");
+		break;
+
+	case LOC_EVT_ERROR:
+		printk("Getting location failed\n\n");
+		break;
+	case LOC_EVT_GNSS_ASSISTANCE_REQUEST:
+		printk("Getting location assistance requested. Not doing anything.\n\n");
+		break;
+	default:
+		printk("Getting location: Unknown event\n\n");
+		break;
+	}
+
+	k_sem_give(&location_event);
+}
+
+static void location_event_wait(void)
+{
+	k_sem_take(&location_event, K_FOREVER);
+}
+
+/**
+ * @brief Retrieve location so that fallback is applied.
+ *
+ * @details This is achieved by setting GNSS as first priority method and giving it too short
+ * timeout. Then a fallback to next method, which is cellular in this example, occurs.
+ */
+static void location_with_fallback_get(void)
+{
+	int err;
+
+	loc_config_defaults_set(&config, 2, methods);
+	loc_config_method_defaults_set(&methods[0], LOC_METHOD_GNSS);
+	/* GNSS timeout is set to 1 second to force a failure. */
+	methods[0].gnss.timeout = 1;
+	loc_config_method_defaults_set(&methods[1], LOC_METHOD_CELLULAR);
+
+	printk("Requesting location with short GNSS timeout to trigger fallback to cellular...\n");
+
+	err = location_request(&config);
+	if (err) {
+		printk("Requesting location failed, error: %d\n", err);
+		return;
+	}
+
+	location_event_wait();
+}
+
+/**
+ * @brief Retrieve location with default configuration.
+ *
+ * @details This is achieved by not passing configuration at all to location_request().
+ */
+static void location_default_get(void)
+{
+	int err;
+
+	printk("Requesting location with the default configuration...\n");
+
+	err = location_request(NULL);
+	if (err) {
+		printk("Requesting location failed, error: %d\n", err);
+		return;
+	}
+
+	location_event_wait();
+}
+
+/**
+ * @brief Retrieve location with GNSS high accuracy.
+ */
+static void location_gnss_high_accuracy_get(void)
+{
+	int err;
+
+	loc_config_defaults_set(&config, 1, methods);
+	loc_config_method_defaults_set(&methods[0], LOC_METHOD_GNSS);
+	methods[0].gnss.accuracy = LOC_ACCURACY_HIGH;
+	methods[0].gnss.num_consecutive_fixes = 3;
+
+	printk("Requesting location with high GNSS accuracy...\n");
+
+	err = location_request(&config);
+	if (err) {
+		printk("Requesting location failed, error: %d\n", err);
+		return;
+	}
+
+	location_event_wait();
+}
+
+#if defined(CONFIG_LOCATION_METHOD_WIFI)
+/**
+ * @brief Retrieve location with WiFi positioning as first priority and cellular as second.
+ */
+static void location_wifi_get(void)
+{
+	int err;
+
+	loc_config_defaults_set(&config, 2, methods);
+	loc_config_method_defaults_set(&methods[0], LOC_METHOD_WIFI);
+	loc_config_method_defaults_set(&methods[1], LOC_METHOD_CELLULAR);
```

Comment:
We only have examples of setting GNSS config options. I think we should show setting Wifi and/or cellular options too. At least set one option somewhere.